projects/fal/src/fal/api.py

```diff
@@ -791,11 +794,26 @@ class BaseServable:
     def collect_routes(self) -> dict[RouteSignature, Callable[..., Any]]:
         raise NotImplementedError
 
-    def _add_extra_middlewares(self, app: FastAPI):
+    def _add_extra_middlewares(self, app: FastAPI) -> None:
         """
         For subclasses to add extra middlewares to the app.
         """
-        pass
+
+        # Common middlewares
```

Comment:
For common stuff, i'd do something like the 404 handler and initialize it directly at the `_build_app()`.